### ISSUE TITLE
[Request] Change required fields

### ISSUE THREAD

**Issue by 2br-2b** [apozas/contactdiary#14]
First off, thank you so much for making this app! I think this is the kind of contact tracing that will be most effective. While Apple/Google's solution may work, it requires a lot of things many people aren't comfortable with, like leaving Bluetooth on 24/7.

My only problem with the app is that sometimes, I don't want to put in all of the information the app currently requires for a contact (event, close contact, relative/close friend, etc.). It's not a matter of security, as that information can't be transfered easily without network access.

The problem is simply one of convenience. As a college student, I want to keep track of who I'm in contact with. I wish I could just enter a friend's name, have the date autofill, and just leave it at that. I don't necessary want to have to enter all of the info each and every time I enter someone's name, especially if I'm entering multiple friends' names. I'd like to enter their name and leave it at that. That is all I need to (or want to) keep track of.

I just want to use this app to make a list of everyone I'm I'm close contact with so I can text them all if I catch covid. I don't always need to know if I was in contact with them inside or outside - if i catch covid, either I'll need to text them or not.

This could either be implemented as an option. Users would choose which fields are required. The easier fix is to only require the person's name, but that allows for less customisability.

**Comment by apozas** [comment]
Hi,

First off, thank you for your feedback. I am happy to know that you are finding Contact Diary useful.

I see what you mean. Contact Diary has two primary goals in mind: to serve you as a log of your contacts, and to have a summary that can be useful if you get contacted by a professional contact tracer. The required options are those that conform the bare minimum that a contact tracer may need to assess the risk of the contacts and advise you and them to take any measures, even if it may be "too much" information for the first goal (you know if you are a close friend of Mary, and even if you have kept a safe distance outdoors you may want to let her know about your positive if you care about her). For now, it is my impression that there are more people that find the required fields useful than people that do not, but I will leave this issue open in the aim of hearing more experiences and opinions.

With this, _I am not saying that this will not be implemented_, but rather than it will not be on the top of the list. In the meantime, let me suggest some workarounds within the current functionalities of the app:

- If you already have a contact with the name you care about, or you have the same contacts under the same conditions recursively (like college or work colleagues), from another day, you can long-press and select "Make a Copy Today" to create a duplicate of that entry in today's register.

- You can also duplicate multiple entries at once by choosing "Select" in the long-press menu, choosing the contacts, and pressing the icon next to the trash can in the top-right corner.

- If you are entering names of multiple friends, you can instead create an event and add all those names in the Companions field. This is what I do whenever I go to office.

I hope at least one of the solutions offered satisfies your needs. As I said, I will leave this issue open to hear other opinions and experiences regarding the use of required fields.

**Comment by GenosseFlosse** [comment]
I'll second @2br-2b 's opinion. I actually have  many entries where location equals the name. The only required field that I personally think are the most useful for me are the distance and the name. Whether someone is my relative or close friend doesn't change occasionally 😂

But I also think about people only wanting to store the list of names and for convenience do not want to store more information. These people tend to be reluctant of this apps usage if forced to set more information fields. 

Since I will notify everyone on my list in case I encounter a covid infection most fields do not matter. I still fill them in but my friend is a bit annoyed of having to put everything in: "when I meet my friend xyz I always know where when and how if I know the date from the app, I don't need to store that". 

So the question is if this information requirement is a special feature on behalf of contact tracers or not. And if so, there should probably be an info box stating just that. Or a pop-up asking you whether you think you will know where you met this person in two weeks from now?

**Comment by 2br-2b** [comment]
I think a popup would be a bit too intrusive unless you could say "Do not show this message again" and then forget that setting later.

I do understand, though, what you're saying. It's just that I know that if I met this person, I'll know where/how I met them. On the other hand, I can still enter the other info; it just takes longer, and the info doesn't always seem necessary to me.

**Comment by apozas** [comment]
Thanks a lot for the thoughts. These are valid suggestions. I want to think about how to best implement it, and I may come back to you with questions about them.

**Comment by ltGuillaume** [comment]
If I fill in the name John and leave the location blank, an option to autofill it with "John's" would be nice. 

Currently, filling in the duration is rather cumbersome. I think it would be better to have separate fields for hours and minutes, with plus and minus buttons next to them, and only an input field opening when you click the number.

**Comment by apozas** [comment]
Hi,

After playing around for a few days with the duration field myself, I liked much more the way of choosing times. I am seriously considering moving back to a "beginning time", followed by an "ending time" in case one wants to specify the duration. Also, I had another realization, this one regarding the required fields. I should not be nor feel responsible for what the users insert into the forms. Therefore, the next version will only have the bare minimum as required, this is, just the contact/event name. The rest will be optional.

**Comment by ltGuillaume** [comment]
@apozas I think this would be the right approach, thanks!

**Comment by 2br-2b** [comment]
Thanks! I think this is the best approach.

Would it be feasible to have an option in the preferences to choose required fields? That way, if a user always wants to always remember to do a certain field, they can.

On the other hand, this is just icing on top and not a required feature. Thank you do much for the change you're already making!

**Comment by apozas** [comment]
> Would it be feasible to have an option in the preferences to choose required fields?
Yes, that is the ideal final form. It will take a bit of time, since first I want to focus on other features that I find more important (import/export, for example). But in the end, it should look like this :)
